### ISSUE TITLE
Re-enable branch protection for main branch

### ISSUE THREAD

**Issue by lionello** [DefangLabs/defang#24]
Protection rules were disabled to allow the Goreleaser task to update the Nix `default.nix` file from the GitHub Action. The action uses the default `$GITHUB_TOKEN` which is for a pseudo-user and cannot be exempted from the branch protection rules. See discussion at https://github.com/orgs/community/discussions/13836

**Comment by lionello** [comment]
Should be possible with Repo Rulesets https://github.com/DefangLabs/defang/settings/rules

**Comment by KevyVo** [comment]
After exploring many options and a lot of discussion we have decided to just remove the support for [GoReleaserbot](https://github.com/goreleaserbot) can generate and [publish a nix derivation](https://goreleaser.com/customization/nix/). The cons out weight the pros to keep the Nix support for goreleaser. 

The options explore were:
- Looking to make a release branch which would not have push protection
- Make a Github bot and use a PAT with Goreleaser
- Make Goreleaser create a PR, which need to be manually merged in

In conclusion we are just going to remove support for Goreleaser to publish a nix derivation and enable branch protection.

More info in this [notion doc](https://www.notion.so/defang-io/Github-Branch-Protection-1c08fc22c43080ca8297f78d381f13ec?pvs=4)
